docs/ReleaseNotes.txt

```diff
@@ -1,3 +1,134 @@
+   `cortex-a78`, `cortex-a510`, and `cortex-a520`, along with support for new
+   Apple devices.
+ - New double-pumped targets have been introduced: `neon-i16x16` and
+   `neon-i8x32`.
+ - Dot product operations are now supported using native ARM instructions
+   (`sdot`/`udot`).
+ - Performance on ARMv8 has improved by an average on 13%.
+
+Generic Targets:
+
+In this release, generic targets were introduced in ISPC. Their main goal is to
+simplify ISPC target management and serve as the foundation for
+hardware-specific targets, requiring only selective tuning when performance
+expectations are not met.
+
+ARM targets have been refactored to use generic targets as a baseline, resulting
+in cleaner code and improved performance. This change also makes it easier to
+add support for new architectures, such as RISC-V or any other LLVM-supported
+target.
+
+Generic targets can also be used as standalone targets in cases where no native
+target exists with the required width for a particular CPU (e.g., a 32-wide
+target for SSE4). This can be done by specifying the following options in ISPC:
+
+`--target=generic-i1x32 --cpu=penryn`
+
+A complete list of all generic targets and the architectures they support can
+be found in the output of:
+
+`ispc --support-matrix`
+
+Code Generation:
+
+ - The `-O1` optimization pipeline has been further optimized for size: loop
+   unrolling and function inlining have been adjusted accordingly.
+ - Improved generated code for the `count_leading_zeros` and
+   `count_trailing_zeros` functions by producing native instructions ( e.g.
+   `vplzcntq`).
+ - Improved generated code for masked load/stores for int8/int16 types on AVX512
+   by generating native instructions (`vmovdqu8`, `vmovdqu16`).
+ - Improved code generation when returning structs from functions by eliminating
+   unnecessary `mov` instructions.
+
+Language Changes:
+
+ - Enhanced support for LLVM intrinsics when the `--enable-llvm-intrinsics` flag
+   is used, including support for intrinsics with no arguments (#3112) and
+   overloaded intrinsics.
+ - Added user-visible macro definitions for the LLVM version that ISPC is based
+   on.
+ - The `__attribute__((deprecated))` attribute can now be applied to functions,
+   generating a warning when the function is called.
+
+Deprecated Targets:
+
+ - The KNL (`avx512knl-x16`) target has been removed.
+
+Compiler Switches Behavior:
+
+ - The `--darwin-version-min` option has been added to specify the minimum
+   deployment target version for macOS and iOS applications. This addresses a
+   new linker behavior introduced in Xcode 15.0, which issues a warning when no
+   version is provided.
+ - The `--nocpp` command-line flag is now deprecated and will be removed in a
+   future release.
+
+Dispatch Behavior:
+
+ - The behavior of user programs when no supported ISA is detected in the
+   auto-dispatch code has changed. Instead of raising the ``SIGABRT`` signal, the
```

Comment:
This paragraph seem to be the only instance when we do double ticks to highlight markdown values in Release Notes. It's just a remark.